pkg/controller/service/nlbv2/loadbalancer.go

```diff
@@ -140,27 +140,25 @@ func (mgr *NLBManager) Update(reqCtx *svcCtx.RequestContext, local, remote *nlbm
 		}
 	}
 
-	needUpdate := false
 	for _, l := range local.LoadBalancerAttribute.ZoneMappings {
-		found := false
+		match := false
 		for _, r := range remote.LoadBalancerAttribute.ZoneMappings {
 			if l.ZoneId == r.ZoneId && l.VSwitchId == r.VSwitchId {
-				found = true
-				break
+				if l.AllocationId == "" || l.AllocationId != "" && l.AllocationId == r.AllocationId {
+					match = true
+					break
+				}
 			}
 		}
-		if !found {
-			needUpdate = true
+		if !match {
```

Comment:
UpdateNLBZones() should be moved outside the for loop to avoid repeated updates

Reply:
Hi @gujingit ,

In my opinion, that code will only run one time(see the below sample).  I'll put that code back if you insist on/prefer it.

```go
for {
  for { }
  if (condition statement) {
       UpdateNLBZones()
       break;
  }
}
```

Reply:
looks good to me